### PR TITLE
TCA Navigation

### PR DESCRIPTION
**Pull request by stephencelis**
This PR upgrades many of our domains to use TCA's navigation APIs. It's not an exhaustive list, since some contenders are not yet easy to transition due to our shared settings state. Once we refactor settings to go through a dependency it should be easier to migrate those.

I've also included a few other modernizations, like using TCA's built-in `onChange` operator over the one we previously built from scratch for this app.

### CODE REVIEW

File: App/isowords.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1339,7 +1339,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 15.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 16.0;
```
Comment (stephencelis) [inline App/isowords.xcodeproj/project.pbxproj:1342]: To use `navigationDestination` I've bumped the deployment target. Up for alternate ideas, though.

File: Sources/AppFeature/AppDelegate.swift [hunk 2]
```diff
@@ -25,92 +25,94 @@ public struct AppDelegateReducer: Reducer {
...
 
-  public func reduce(into state: inout State, action: Action) -> Effect<Action> {
-    switch action {
-    case .didFinishLaunching:
-      return .run { send in
-        await withThrowingTaskGroup(of: Void.self) { group in
-          group.addTask {
-            for await event in self.userNotifications.delegate() {
-              await send(.userNotifications(event))
+  public var body: some ReducerOf<Self> {
```
Comment (stephencelis) [inline Sources/AppFeature/AppDelegate.swift:28]: While some of these `reduce(into:)` needed to migrate to the `body` style to compose with `ifLet`, I've migrated all of them for consistency.

File: App/isowords.xcodeproj/project.pbxproj (hunk 1 above)
Comment (mbrandonw) [reply App/isowords.xcodeproj/project.pbxproj:1342]: nah iOS 16 sounds good to me!

File: Sources/GameCore/GameCore.swift [hunk 3]
```diff
@@ -17,16 +17,48 @@ import TcaHelpers
...
+    }
+    public var body: some ReducerOf<Self> {
+      Scope(state: /State.gameOver, action: /Action.gameOver) {
+        GameOver()
+      }
+      Scope(state: /State.upgradeInterstitial, action: /Action.upgradeInterstitial) {
+        UpgradeInterstitial()
+      }
+    }
+  }
```
Comment (stephencelis) [inline Sources/GameCore/GameCore.swift:52]: `Game` and `Home` have some of the more interesting navigation refactors. Here we're able to house 4 destinations.

**Pull request merged**